src/io.rs

```diff
@@ -50,3 +70,27 @@ pub fn read_embeddings_view(
 
     Ok(embeds?)
 }
+
+pub fn read_quantized_embeddings(
```

Comment:
Change 9: This function read any embeddings before. So, it will 'reconstruct' embedding matrices that are not quantized. In order to avoid this, I specifically have to read the embeddings as Embeddings<VocabWrap, QuantizedArray> or Embeddings<VocabWrap, MmapQuantizedArray>.Then I convert them into Embeddings<VocabWrap, StorageWrap> in order to be able to represent both in-memory and mmap'ed arrays. But then at least the command will fail when trying to apply reconstruction on non-quantized embredings.